src/worker.cpp

```diff
@@ -1185,4 +1183,277 @@ void Worker::collectFrusFromJson()
         }
     }
 }
+
+types::BackupMapVariant Worker::getBackupMap()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        const std::string& l_backupMap =
+            m_parsedJson["backupAndRestore"].value("BackupMap", "");
+        if (constants::VpdBackupMap.find(l_backupMap) !=
+            constants::VpdBackupMap.end())
+        {
+            return constants::VpdBackupMap.at(l_backupMap);
+        }
+    }
+    return std::monostate{};
+}
+
+bool Worker::isBackupAndRestoreSourceOnCache(const std::string& i_pathName)
+{
+    if (i_pathName.compare("source1") == constants::STR_CMP_SUCCESS &&
+        m_parsedJson.contains("backupAndRestore"))
+    {
+        if (m_parsedJson["backupAndRestore"].contains("Source1Path") &&
+            m_parsedJson["backupAndRestore"]["Source1Path"].contains(
+                "InventoryPath"))
+        {
+            return true;
+        }
+    }
+    else if (i_pathName.compare("source2") == constants::STR_CMP_SUCCESS &&
+             m_parsedJson.contains("backupAndRestore"))
+    {
+        if (m_parsedJson["backupAndRestore"].contains("Source2Path") &&
+            m_parsedJson["backupAndRestore"]["Source2Path"].contains(
+                "InventoryPath"))
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+std::string Worker::getBackupAndRestorePath(const std::string& i_pathName)
```

Comment:
I did not understand this implementation. Why so complex?

getBackupAndRestorePath(const std::string& i_pathName); can be modified as, taking into consideration the JSON update that has been suggested above.


getBackupRestorePath(const std::string i_pathName, std::string& o_location)
{
//lets say tag passed is "source1"
if(m_parsedJson["backupAndRestore"].contains(i_pathName))
{
   if(m_parsedJson["backupAndRestore"][i_pathName].contains(hwPath))
  {
       location = "hardware"; //or whatever relevant you want.
   }
   else if(m_parsedJson["backupAndRestore"][tag].contains(invPath))
   {
        location = "inventory"; //or whatever relevant you want.
    }
}
}